Fwupd test suite

philipanda opened the pull request:
*ref: ncm-1922*

philipanda committed:
dasharo-compatibility/fwupd-support.robot: Add suite

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
novacustom-common.robot: Add Capsule udpate support

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
[REMOVE ME] Add temporary desk configuration

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
fwupd-support.robot: Add Fedora cases

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
test_cases.json: Add FWUPD cases

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
fwupd-support.robot: Add QubesOS cases

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
test_cases.json: Add FWUPD QubesOS cases

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
fwupd-support.robot: Ensure ME disabled

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
fwupd-support: Move LVS test cases into separate suite

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

macpijan commented:
@BeataZdunczyk @philipanda we are missing a reviewer here

SergiiDmytruk commented on dasharo-compatibility/fwupd-support.robot:
```diff
@@ -0,0 +1,119 @@
+    END
+
+    Should Contain    ${out}
+    ...    @{devices}
```
nit: does this need to be on a separate line?

philipanda replied:
No, not sure why I did it that way

philipanda replied:
https://github.com/Dasharo/open-source-firmware-validation/pull/974/commits/1b4ce8f1b7801f53ccabc9cfe7442b79767c81ba

SergiiDmytruk commented on util/fwupd-support-lvfs.robot:
```diff
@@ -0,0 +1,101 @@
+    ...    Password=${password}
+    ...    separator=${EMPTY}
+
+    Execute Command In Terminal    touch /etc/fwupd/remotes.d/3mdeb-embargo.conf
```
Why `touch file` before doing `printf > file`?

philipanda replied:
Yes, its pointless

philipanda replied:
https://github.com/Dasharo/open-source-firmware-validation/pull/974/commits/f7f117f58984800fc01cc8845da2d5fa929ca164

SergiiDmytruk commented on util/fwupd-support-lvfs.robot:
```diff
@@ -0,0 +1,101 @@
+
+    Switch To Root User
+    Setup Fwupd Embargo Config Linux    ${username}    ${password}
+    Execute Command In Terminal    printf '[fwupd]\\nOnlyTrusted=true\\n' | sudo tee /etc/fwupd/fwupd.conf
```
Already root user.
```suggestion
    Execute Command In Terminal    printf '[fwupd]\\nOnlyTrusted=true\\n' > /etc/fwupd/fwupd.conf
```

philipanda replied:
Good catch. I've been testing it manually as a normal user and more or less copied the required commands, but we switch to root before in the test.

philipanda replied:
https://github.com/Dasharo/open-source-firmware-validation/pull/974/commits/5232dc59c47f8c35e1a907b6fa3f09460b0b650b

SergiiDmytruk commented on util/fwupd-support-lvfs.robot:
```diff
@@ -0,0 +1,101 @@
+    Execute Command In Terminal    printf "${embargo_config}" > /etc/fwupd/remotes.d/3mdeb-embargo.conf
+
+Clean Up Fwupd Embargo Config Linux
+    Execute Command In Terminal    rm -rf /etc/fwupd/remotes.d/3mdeb-embargo.conf
```
Since we expect the file to be there and it must be a file, no need for `-rf`:
```suggestion
    Execute Command In Terminal    rm /etc/fwupd/remotes.d/3mdeb-embargo.conf
```

philipanda replied:
`-f` is useful, because on some systems an interactive prompt could appear on the terminal asking for confirmation. `-f` skips that. `-r` is unnecessary and we can remove it if you prefer it to be cleaner

philipanda replied:
https://github.com/Dasharo/open-source-firmware-validation/pull/974/commits/708ee93ef0e3c31cf1badc79bab76b4b00ebe202

SergiiDmytruk replied:
There is also `--interactive=never`, but `-f` should work even with crippled implementation like busybox as well.  My problem with things like `-r` is that they can hide mistakes from other places.

SergiiDmytruk commented on variables.robot:
```diff
@@ -273,6 +273,11 @@ ${OS_UBUNTU}=               ubuntu
+&{RTE74}=                   ip=192.168.10.185
+...                         platform=novacustom-v540tu
+...                         platform_vendor=Clevo
+...                         sonoff_ip=192.168.4.229
```
Looks like the IP has changed:
```suggestion
...                         sonoff_ip=192.168.10.148
```

And maybe you didn't intend to keep it? Commit's title is "[REMOVE ME] Add temporary desk configuration".

philipanda replied:
Yes, OSFV doesn't really help if you want to test on a custom configuration on your desk. The `variables.robot` must be modified even if the values there are not really helpful. OSFV now requires running on a clean git tree so that the results uploaded during tests are collected from a known revision that exists on the remote. This commit should be reverted after the testing is done.

philipanda replied:
https://github.com/Dasharo/open-source-firmware-validation/pull/974/commits/3e3d67f288f7e3ce26c24cfc3e93841354f38d9c

SergiiDmytruk commented on dasharo-compatibility/fwupd-support.robot:
```diff
@@ -0,0 +1,119 @@
+    [Documentation]    Test if the supported hardware is properly detected
+    ...    by fwupd
+    [Tags]    semiauto
+    Execute Manual Step    Power on and boot into QubesOS
```
Why manual steps?  If that's because of default state not being usable remotely, that can be changed with some script like installation via openQA does that to get a system with serial and SSH to root in dom0.  Maybe not for this particular PR, I'm just wondering why it's manual here.

philipanda replied:
Yes, it could be automated, we just don't have support for automatic Qubes tests on OSFV right now and this PR works with what is available.
Like you said, it would probably require running the installer with a custom kickstart config, so https://github.com/dasharo/preseeds should be a good place to start with this. Maybe the kickstart from OpenQA could be mostly copied there? If serial console and SSH would be available on QubesOS it would help with its tests greatly. @BeataZdunczyk what do you think about automating QubesOS tests? That way we wouldn't have to waste time on creating the documentation for manual tests.

SergiiDmytruk replied:
> Maybe the kickstart from OpenQA could be mostly copied there?

Yes, I think so.  I was thinking of reworking it into a script to run after installing Qubes OS for tests, but installing it with all the necessary changes is even better.

Resolving the comment as it's not specific to this PR.

philipanda committed:
fwupd-support.robot: remove unnecessary line break

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
Revert "[REMOVE ME] Add temporary desk configuration"

This reverts commit 84d800e3033d2c030535a8d84e91ddd1582ca07c.

philipanda committed:
fwupd-support-lvfs: Remove -r flag from rm call

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
fwupd-support-lvfs: Remove pointless touch lvfs remote file

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
fwupd-support-lvfs: Remove sudo tee when we already are root

Co-authored-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>
Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

filipleple reviewed (approved):
Overall LGTM, but CI is failing (not sure if it's for a valid reason) and comments need to be resolved first

SergiiDmytruk committed:
scripts/ci/develop_pr_auto_regression.sh: don't check nonexistent run 0

The loop was using a 0-based index on an 1-based array.  Enumerate keys
and query values per key instead.

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk reviewed (approved):
> CI is failing (not sure if it's for a valid reason)

There are run 1 and run 2, but:
```
Run 0 failed with exit code 1. Check ./logs/run_0.log for details.
```

Sent https://github.com/Dasharo/open-source-firmware-validation/pull/1041 which should fix this.  Later rebase this PR on top of the new one, so I guess we'll see whether it helps.

SergiiDmytruk commented:
CI is still failing because MSI just doesn't seem to boot, so it's not really a test failure in this case.

philipanda commented:
> There are run 1 and run 2, but:
>
> Run 0 failed with exit code 1. Check ./logs/run_0.log for details.
>
> Sent https://github.com/Dasharo/open-source-firmware-validation/pull/1041 which should fix this. Later rebase this PR on top of the new one, so I guess we'll see whether it helps.

I am sure I've been fixing this already... Not sure what happened with my changes and I guess there is no reason to look for them now. Thanks for noticing this
